### PR TITLE
quell type and immutable deprecation warnings under 0.7

### PR DESCRIPTION
**Pull request by Sacha0**
Wonderful meeting you last week `@jrevels`! :)

### CODE REVIEW

**Comment by Sacha0** [comment]
(The CI failures on 0.4 and 0.5 are of course expected, and the macOS nightly failure seems unrelated. Best!)

**Comment by tkelman** [comment]
should raise minimum julia version in REQUIRE to 0.6 then?

**Comment by Sacha0** [comment]
Yes, assuming `@jrevels` is on board?

**Comment by jrevels** [comment]
Sorry for the late response, thanks for this! 

> should raise minimum julia version in REQUIRE to 0.6 then?

Seems like a good idea.

**Comment by Sacha0** [comment]
Done, no worries, my pleasure, and preemptive happy 4th! :)

File: REQUIRE [hunk 1]
```diff
@@ -1,3 +1,3 @@
-julia 0.4
+julia 0.6
```
Comment (tkelman) [inline REQUIRE]: and drop earlier from ci

Comment (Sacha0) [inline REQUIRE]: Thanks! Let's see whether I fixed that correctly...

Comment (tkelman) [inline REQUIRE]: looks right

**Comment by Sacha0** [comment]
Thanks `@jrevels`! :)

**Comment by tkelman** [comment]
I don't think this works correctly when the consumer doesn't use Compat itself. Maybe need to interpolate it into the macro expansion?

**Comment by Sacha0** [comment]
> I don't think this works correctly when the consumer doesn't use Compat itself. Maybe need to interpolate it into the macro expansion?

(Did you mean to post this somewhere else? :) )

**Comment by tkelman** [comment]
No, see the JuliaCIBot failures downstream on the tag pr

**Comment by tkelman** [comment]
ah no whichever pr added at-module calls sorry

**Pull request merged**